task/backend/coordinator/task_coordinator.go

```diff
@@ -0,0 +1,185 @@
+package coordinator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/task/backend"
+	"github.com/influxdata/influxdb/task/backend/executor"
+	"github.com/influxdata/influxdb/task/backend/middleware"
+	"github.com/influxdata/influxdb/task/backend/scheduler"
+	"go.uber.org/zap"
+)
+
+var _ middleware.Coordinator = (*Coordinator)(nil)
+var _ Executor = (*executor.TaskExecutor)(nil)
+var _ scheduler.SchedulableService = (*SchedulableTaskService)(nil)
+
+// Executor is an abstraction of the task executor with only the functions needed by the coordinator
+type Executor interface {
+	ManualRun(ctx context.Context, id scheduler.ID, runID scheduler.ID) (executor.Promise, error)
+	Cancel(ctx context.Context, runID scheduler.ID) error
+}
+
+type TaskCoordinator struct {
+	logger *zap.Logger
+	sch    scheduler.Scheduler
+	ex     Executor
+	// UpdateTaskFunc
+
+	limit int
+}
+
+type OptionC func(*TaskCoordinator)
+
+// SchedulableTask is a wrapper around the Task struct, giving it methods to make it compatible with the Scheduler
+type SchedulableTask struct {
+	*influxdb.Task
+}
+
+// SchedulableTaskService provides an API to update the LatestScheduled time of a task
+type SchedulableTaskService struct {
+	ts influxdb.TaskService
```

Comment:
currently wrapping the whole task service, but would like to just include the method necessary for the coordinator (which is UpdateTask)